### ISSUE TITLE
TS 4.1.2 Cannot find name 'FieldValue' (solved in 4.2)

### ISSUE THREAD

**Issue by sk-** [microsoft/TypeScript#41629]
In TS 4.1.2, depending on how you import `FieldValue` (a firestore type), the class won't be defined. This affects only 4.1.2, as it seems to have been fixed in 4.2.

**TypeScript Version:**  4.1.2

**Search Terms:**
FieldValue undefined cannot

**Code**

```js
const {FieldValue} = require('firebase-admin').firestore;
console.info(FieldValue);
```

**Expected behavior:**
No error is raised.

**Actual behavior:**
```
 Cannot find name 'FieldValue'.
```

Note that the code correctly prints out:

```
[class FieldValue]
```

Note also that the following code compiles without errors:
```js
const admin = require('firebase-admin');
const {FieldValue} = admin.firestore;
console.info(FieldValue);

const {equal} = require('assert').strict
console.info(equal);
```
**Playground Link:** 

**Related Issues:**

**Comment by andrewbranch** [comment]
@sandersn, does this look like something you intentionally fixed for 4.2? It is a pretty bizarre error. Would it be a candidate to backport?

**Comment by sk-** [comment]
I bisected it and the error was introduced in `4.1.0-dev.20200818` and fixed in `4.2.0-dev.20201112`.

**Comment by sk-** [comment]
It seems that this was fixed by https://github.com/microsoft/TypeScript/pull/41479 and introduced by https://github.com/microsoft/TypeScript/pull/39770.

**Comment by andrewbranch** [comment]
Thanks 🕵️! That makes this a duplicate of #41422. We’ll discuss putting it in the next patch release.
